test/test-search.ts

```diff
@@ -0,0 +1,105 @@
+      const result1 = await search.search(content1);
+      t.true(result1.requiredModules.has('http'));
+
+      const content2 = 'const a = require(`http`);';
+      const result2 = await search.search(content2);
+      t.true(result2.requiredModules.has('http'));
+    });
+
+test('searchValue should have http and fs module', async t => {
+  const content =
+      'const a = require(\'http\');\n const b = require(`${\'fs\'}`);';
+  const result = await search.search(content);
+  t.true(result.requiredModules.has('http'));
+  t.true(result.requiredModules.has('fs'));
+});
+
+test('line number is accurate for search', async t => {
+  const content =
+      'const a = require(\'http\');\n const b = require(`${\'fs\'}`);';
+  const result = await search.search(content);
+
+  const httpPosition = (result.requiredModules.get('http'));
+  
+  if(httpPosition !== undefined && assert(httpPosition !== null, 'position of http is null') ){
+    t.deepEqual(httpPosition.lineStart, 1);
+  }
+
+  const fsPosition = (result.requiredModules.get('fs'));
+  //console.log('HEKWHJFKJSFKBFSBF fs: ' + fsPosition);
+  /*if(t.true(assert(fsPosition === null && fsPosition !== undefined))){
+    t.deepEqual(fsPosition.lineStart, 2);
+  }*/
+});
+
+test(
+    'search function should return correct dynamic eval position with ' +
+        'require arg as a concatenation of strings that forms http',
+    async t => {
+      const content = 'const a = require(\'h\' + \'t\' + \'t\' + \'p\');';
+      const result = await search.search(content);
+      if (assert(result.dynamicEvals[0], 'position does not exist in array')) {
+        t.deepEqual(result.dynamicEvals[0].lineStart, 1);
+      }
+    });
+
+test(
+    'search function should return correct dynamic eval position with ' +
+        'require arg as a substring that forms http',
+    async t => {
+      const content1 =
+          `const a = \'anotherhttp\'\nconst b = require(a.substring(6));`;
+      const result1 = await search.search(content1);
+      if (assert(result1.dynamicEvals[0], 'position does not exist in array')) {
+        t.deepEqual(result1.dynamicEvals[0].lineStart, 2);
+      }
+
+      const content2 = 'const a = require(\'anotherhttp\'.substring(6))';
+      const result2 = await search.search(content2);
+      if (assert(result2.dynamicEvals[0], 'position does not exist in array')) {
+        t.deepEqual(result2.dynamicEvals[0].lineStart, 1);
+      }
+    });
+
+test(
+    'search function should return correct dynamic eval position with ' +
+        'require arg as a function that returns http',
+    async t => {
+      const content =
+          'function returnHttp(){return \'http\';}\nconst a = require(returnHttp);';
+      const result = await search.search(content);
+      if (assert(result.dynamicEvals[0], 'position does not exist in array')) {
+        t.deepEqual(result.dynamicEvals[0].lineStart, 2);
+      }
+    });
+
+/*test('require callee is defined as a variable', async t => {
```

Comment:
Can you either uncomment or remove commented-out code?

Reply:
done